testsuite/basic/src/test/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParserTest.java

```diff
@@ -35,6 +40,28 @@ public void testParse() throws Exception {
         assertNotNull(context);
     }
 
+    @Test
+    public void testParseJwtSignedWith1024RsaKeyLengthAllowed() throws Exception {
+        KeyPair pair = KeyUtils.generateKeyPair(1024);
+        String jwt = TokenUtils.generateTokenString(pair.getPrivate(), "kid", "/Token1.json", null, null);
+        JWTAuthContextInfo context = new JWTAuthContextInfo((RSAPublicKey) pair.getPublic(), "https://server.example.com");
+        context.setRelaxVerificationKeyValidation(true);
+        assertNotNull(parser.parse(jwt, context).getJwtClaims());
+    }
+
+    @Test(expected = InvalidJwtException.class)
+    public void testParseJwtSignedWith1024RsaKeyLengthDisallowed() throws Throwable {
+        KeyPair pair = KeyUtils.generateKeyPair(1024);
+        String jwt = TokenUtils.generateTokenString(pair.getPrivate(), "kid", "/Token1.json", null, null);
+        JWTAuthContextInfo context = new JWTAuthContextInfo((RSAPublicKey) pair.getPublic(), "https://server.example.com");
+        try {
+            parser.parse(jwt, context);
+            fail("InvalidJwtException is expected");
+        } catch (ParseException ex) {
+            throw ex.getCause();
+        }
```

Comment:
Consider using `assertThrows` instead of `@Test(expected = ...)`. It's the more JUnit 5 way of doing things, which is probably coming in the future.

```java
ParseException thrown = assertThrows(ParseException.class, () -> parser.parse(jwt, context));
assertTrue(thrown.getCause() instanceof InvalidJwtException);
```

Reply:
Parent already has JUnit 5 support, so we could move our tests there.

Reply:
@MikeEdgar @radcortez `assertThrows` is good, this is definitely cleaner/more compact, done